daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala

```diff
@@ -288,6 +288,21 @@ private final class Validation[Nid, Cid](implicit ECid: Equal[Cid]) {
                   keyIsReplayedBy(key1, key2) &&
                   result1 === result2 =>
               loop(rest1, rest2, stack)
+            case (
+                  Node.NodeRollback(
+                    children1,
+                    version1,
+                  ),
+                  Node.NodeRollback(
+                    children2,
+                    version2,
+                  ),
+                ) if version1 == version2 =>
+              loop(
+                children1.iterator.to(LazyList),
+                children2.iterator.to(LazyList),
+                stack,
+              )
             case _ =>
               Left(ReplayNodeMismatch(recorded, nid1, replayed, nid2))
```

Comment:
I'm sad the scala compiler didn't discover statically that the case of `Rollback` nodes was missing. Initially I though the blame was our this catch-all...
```
            case _ =>
              Left(ReplayNodeMismatch(recorded, nid1, replayed, nid2))
```

And I wondered if being more precise would have provoked the desired unmatched case warnings when the new node type was added....

But no. There are no warnings even if the catch-all is removed. Scala is broken  :(